### PR TITLE
Use same mechanism to manage/load check rules

### PR DESCRIPTION
**Pull request by zippy1978**
This PR is a reworked version of this initial PR: https://github.com/green-code-initiative/ecoCode-ios/pull/16

It fixes https://github.com/green-code-initiative/ecoCode-ios/issues/15

### CODE REVIEW

**Comment by dedece35** [comment]
Hi @zippy1978,
could you correct the new Code Smell please ?

File: pom.xml.versionsBackup [hunk 1]
```diff
@@ -0,0 +1,205 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (dedece35) [inline pom.xml.versionsBackup:1]: useless file for me

File: sonar-ios-plugin/pom.xml.versionsBackup [hunk 2]
```diff
@@ -0,0 +1,70 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (dedece35) [inline sonar-ios-plugin/pom.xml.versionsBackup:1]: useless file for me

File: swift-lang/pom.xml.versionsBackup [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (dedece35) [inline swift-lang/pom.xml.versionsBackup:1]: useless file for me

**Review by dedece35 — CHANGES_REQUESTED** [review]
Hi @zippy1978,
there is a lot of modification and I saw that core system is upgraded.
If you tested this PR with your real test project and it was ok, thus I will approve the PR (after checking my review comments :p )

File: commons-ios/pom.xml.versionsBackup [hunk 4]
```diff
@@ -0,0 +1,34 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
[the remark this answers is not in the document]
Comment (zippy1978) [reply commons-ios/pom.xml.versionsBackup:1]: It's a mistake I remove it

File: commons-ios/src/main/java/io/ecocode/ios/Const.java [hunk 5]
```diff
@@ -15,23 +15,11 @@
  * You should have received a copy of the GNU General Public License
  * along with this program. If not, see <http://www.gnu.org/licenses/>.
  */
-package io.ecocode.ios.rules;
+package io.ecocode.ios;
 
-public final class RepositoryRuleDebt {
+public class Const {
```
[the remark this answers is not in the document]
Comment (zippy1978) [reply commons-ios/src/main/java/io/ecocode/ios/Const.java:20]: Nice !
I change it to an interface

File: swift-lang/pom.xml.versionsBackup (hunk 3 above)
Comment (zippy1978) [reply swift-lang/pom.xml.versionsBackup:1]: It's a mistake I remove it

File: pom.xml.versionsBackup (hunk 1 above)
Comment (zippy1978) [reply pom.xml.versionsBackup:1]: It's a mistake I remove it

**Review by dedece35 — APPROVED** [review]

**Comment by dedece35** [comment]
Hi @zippy1978,
you can now merge this PR :p

**Pull request merged**